site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
+                updated_name["props"][prop] = user_prop
+                updated = True
+
+        # add the email as props
+        if user.email and user.email != name.get("props", {}).get("email"):
+            if "props" not in updated_name:
+                updated_name["props"] = {}
+            updated_name["props"]["email"] = user.email
+            updated = True
+
+        return updated_name if updated else None
+
+    def _update_name(user, name_dict, is_orcid=False):
+        """Updates the name with the new values."""
+        updated_name = _check_if_update_needed(user, name_dict, is_orcid)
+        if updated_name:
+            try:
+                service.update(system_identity, name_dict.get("id"), updated_name)
+            except ValidationError as e:
+                current_app.logger.error(f"Error updating name for user {user.id}: {e}")
+
+    def _create_new_name(user):
+        name = {"id": str(user.id), "props": {"is_cern": True}}
+
+        if user.user_profile.get("given_name"):
+            name["given_name"] = user.user_profile.get("given_name")
+        if user.user_profile.get("family_name"):
+            name["family_name"] = user.user_profile.get("family_name")
+        if user.user_profile.get("affiliations"):
+            name["affiliations"] = [{"name": user.user_profile.get("affiliations", "")}]
+        if user.user_profile.get("orcid"):
+            name["identifiers"] = [
+                {"scheme": "orcid", "identifier": user.user_profile.get("orcid")}
+            ]
+        for prop in prop_values:
+            if user.user_profile.get(prop):
+                name["props"][prop] = user.user_profile.get(prop)
+        try:
+            service.create(system_identity, name)
+        except ValidationError as e:
+            current_app.logger.error(f"Error creating name for user {user.id}: {e}")
+
+    users = []
+    # Allows to sync a single user
+    if user_id:
+        user = User.query.get(user_id)
+        users = [user] if user else []
+    else:
+        users = User.query.filter(User.updated > since).all()
+    name_dict = None
+    for user in users:
+        orcid = None
+        try:
+            name_dict = service.read(system_identity, str(user.id)).to_dict()
+            _update_name(user, name_dict)
+        except NoResultFound:
+            orcid = user.user_profile.get("orcid")
+            if orcid:
+                try:
+                    # Check if the ORCID value is already present and update it
+                    name_dict = service.read(system_identity, orcid).to_dict()
+                    _update_name(user, name_dict, is_orcid=True)
+                except NoResultFound:
+                    _create_new_name(user)
+            else:
+                _create_new_name(user)
+
+
+
+@shared_task()
+def merge_duplicate_names_vocabulary(since=None):
+    """Merges duplicate names in the names vocabulary."""
+    service = current_service_registry.get("names")
```

Comment:
just for readability
```suggestion
    names_service = current_service_registry.get("names")
```